inventory/src/main/java/org/flyve/inventory/categories/Cameras.java

```diff
@@ -76,6 +89,132 @@ public Cameras(Context xCtx) {
+     */
+    public String getFacingState(CameraCharacteristics characteristics) {
+        String value = "N/A";
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            Integer facing = characteristics.get(CameraCharacteristics.LENS_FACING);
+            if (facing != null) {
+                switch (facing) {
+                    case 0:
+                        value = "FRONT";
+                        break;
+                    case 1:
+                        value = "BACK";
+                        break;
+                    case 2:
+                        value = "EXTERNAL";
+                        break;
+                }
+            }
+        }
+        return value;
+    }
+
+    /**
+     * Whether this camera device has a flash unit.
+     * @param characteristics
+     * @return String 0 no available, 1 is available
+     */
+    public String getFlashUnit(CameraCharacteristics characteristics) {
+        String value = "N/A";
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            Boolean bool = characteristics.get(CameraCharacteristics.FLASH_INFO_AVAILABLE);
+            return bool != null ? bool ? "1" : "0" : "N/A";
+        }
+        return value;
+    }
+
+    public Category getCategoryImageFormat(CameraCharacteristics chars) {
+        Category category = new Category("IMAGEFORMAT", "imageformat");
+        for (String imageFormat : getImageFormat(chars)) {
+            category.put("FORMAT", new CategoryValue(imageFormat, "FORMAT", "format"));
+        }
+        return category;
+    }
+
+    /**
+     * version information about the camera device
+     * The available stream configurations that this camera device supports
+     * @param characteristics
+     * @return String The camera device faces the same direction as the device's screen
+     */
+    public ArrayList<String> getImageFormat(CameraCharacteristics characteristics) {
+        ArrayList<String> types = new ArrayList<>();
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            StreamConfigurationMap configurationMap = characteristics.get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP);
+            if (configurationMap != null) {
+                int[] outputFormats = configurationMap.getOutputFormats();
+                if (outputFormats != null) {
+                    for (int value : outputFormats) {
+                        String type = typeFormat(value);
+                        if (type != null) {
+                            types.add(removeCharacters(type));
+                        }
+                    }
+                }
+            }
+        }
+        return types;
+    }
+
+    private String removeCharacters(String type) {
+        return type.replaceAll("[<>]", "");
+    }
+
+    private String typeFormat(int i) {
```

Comment:
@Ivans51 These type format should include in the assets folder as JSON.

Reply:
I'll create new issue about that

Reply:
Seems does not need it, because it does not change in the time.
Take a look if you can use more resources to simplify the code.
https://developer.android.com/reference/android/graphics/ImageFormat

Reply:
The ImageFormat class is used to get information mainly with API camera is deprecated